lib/lita/dependency_update_builder.rb

```diff
@@ -0,0 +1,89 @@
+require "lita/project_repo"
+
+module Lita
+  class DependencyUpdateBuilder
+
+    # This is the number of seconds to wait before re-submitting a build that
+    # is unchanged from the last one we submitted.
+    #
+    # When the build fails after updating dependencies, it could be caused
+    # either by the dependency updates (i.e., a bug) or some kind of
+    # intermittent issue in the build (network failure, upstream service is
+    # down, etc.). Therefore we *do* want to retry the build, but we want to
+    # rate-limit so we're not spamming the build system with something that can
+    # never work.
+    QUIET_TIME_S = 24 * 60 * 60
+
+    attr_reader :repo_url
+    attr_reader :dependency_branch
+    attr_reader :dependency_update_command
+
+    def initialize(repo_url:, dependency_branch:, dependency_update_command:)
```

Comment:
These are required keyword arguments. We haven't been able to use them in Chef because we support versions of ruby that don't have them, but I find them to be pretty optimal for readability and stability at the price of a bit of verbosity.